pkg/tests/ossm/smcp_must_gather_test.go

```diff
@@ -40,18 +45,171 @@ func TestMustGather(t *testing.T) {
 		app.InstallAndWaitReady(t, app.Bookinfo(ns))
 
 		image := env.GetMustGatherImage()
-
-		t.LogStepf("Capture must-gather using image %s", image)
 		dir := shell.CreateTempDir(t, "must-gather-")
-		shell.Executef(t, `mkdir -p %s; oc adm must-gather --dest-dir=%s --image=%s`, dir, dir, image)
-
-		t.LogStep("Check cluster-scoped openshift-operators.servicemesh-resources.maistra.io.yaml")
-		pattern := dir + "/*must-gather*/cluster-scoped-resources/admissionregistration.k8s.io/mutatingwebhookconfigurations/openshift-operators.servicemesh-resources.maistra.io.yaml"
-		matches, err := filepath.Glob(pattern)
-		if err == nil && len(matches) != 0 {
-			t.LogSuccessf("file exists: %s", matches)
-		} else {
-			t.Fatalf("openshift-operators.servicemesh-resources.maistra.io.yaml file not found: %s", matches)
+
+		t.LogStepf("Capture must-gather using image %s without namespace flag", image)
+		output := shell.Executef(t, `mkdir -p %s; oc adm must-gather --dest-dir=%s --image=%s`, dir, dir, "registry-proxy.engineering.redhat.com/rh-osbs/openshift-service-mesh-istio-must-gather-rhel8:2.4.0-11")
+		if strings.Contains(output, "ERROR:") {
+			t.Fatalf("Error was found during the execution of must-gather: %s\n", output)
 		}
+
+		t.NewSubTest("dump files and proxy stats files exist").Run(func(t test.TestHelper) {
+			t.LogStep("Check files exist under the directory of mustgather: openshift-operators.servicemesh-resources.maistra.io.yaml, debug-syncz.json, config_dump_istiod.json, config_dump_proxy.json, proxy_stats")
+			fileList := []string{"version",
+				"openshift-operators.servicemesh-resources.maistra.io.yaml",
+				"debug-syncz.json", "config_dump_istiod.json",
+				"config_dump_proxy.json",
+				"proxy_stats"}
+			verifyFilesExist(t, dir, fileList)
+
+			t.LogStep("verify content of proxy_stats")
+			t.Log("verify that proxy stats file is not empty and contains parameters like: server.stats_recent_lookups, server.total_connections, server.uptime, server.version")
+			proxyStatsFilePath := getPathToFile(t, dir, "proxy_stats")
+			shell.Execute(t,
+				fmt.Sprintf("cat %s", proxyStatsFilePath),
+				assert.OutputContains("server.stats_recent_lookups",
+					"server.stats_recent_lookups is on the proxy_stats file",
+					"server.stats_recent_lookups is not on the proxy_stats file"),
+				assert.OutputContains("server.total_connections",
+					"server.total_connections is on the proxy_stats file",
+					"server.total_connections is not on the proxy_stats file"),
+				assert.OutputContains("server.uptime",
+					"server.uptime is on the proxy_stats file",
+					"server.uptime is not on the proxy_stats file"))
+		})
+
+		t.NewSubTest("version file exist").Run(func(t TestHelper) {
+			t.LogStep("verify version exist")
+			fileList := []string{"version"}
```

Comment:
You already check this in the first subtest.

Reply:
Move the validation of the version file to the first subtest case